tests/integration/test_core.py

```diff
@@ -0,0 +1,124 @@
+# Copyright 2022 Canonical Ltd.
+# Licensed under the GPLv3, see LICENCE file for details.
+
+import secrets
+
+import pytest
+import pytest_operator.plugin
+import requests
+import swiftclient
+import swiftclient.exceptions
+import swiftclient.service
+
+
+@pytest.mark.asyncio
+@pytest.mark.abort_on_fail
+async def test_build_and_deploy(
+    ops_test: pytest_operator.plugin.OpsTest, content_cache_image
+):
+    """
+    arrange: no pre-condition.
+    act: build charm using charmcraft and deploy charm to test juju model.
+    assert: building and deploying should success and status should be "blocked" since the
+        database info hasn't been provided yet.
+    """
+    my_charm = await ops_test.build_charm(".")
+    await ops_test.model.deploy(
+        my_charm,
+        resources={"content-cache-image": content_cache_image},
+        series="jammy",
+    )
+    await ops_test.model.wait_for_idle()
```

Comment:
Should there be an assert at all?

Reply:
since it is just the build_and_deploy test, it is not necessary (and will be moved into a fixture anyway)